core/ts/MaciState.ts

```diff
@@ -549,47 +560,42 @@ class Poll {
             this.copyStateFromMaci()
         }
 
-        const deactivatedKeyHash = hash3([...deactivatedPublicKey.asArray(), salt]);
-        const deactivatedKeyIndex = this.deactivatedKeyEvents.findIndex(d => d.keyHash === deactivatedKeyHash);
+        const deactivatedKeyHash: BigInt = hash3([...deactivatedPublicKey.asArray(), salt]);
+        const deactivatedKeyIndex = this.deactivatedKeyEvents.findIndex(d => d.keyHash.toString() == deactivatedKeyHash.toString());
 
         if (deactivatedKeyIndex === -1) {
-            console.log("Key index is -1");
+            console.log("Deactivated key is missing from the deactivated key events collection from contract");
             return {};
         }
 
         const deactivatedKeyEvent = this.deactivatedKeyEvents[deactivatedKeyIndex];
 
-        const stateIndexInt = parseInt(stateIndex.toString());
-        const computedStateIndex = stateIndexInt > 0 && stateIndexInt <= this.numSignUps ? stateIndexInt - 1 : -1;
-        const currentStateLeaves = [this.stateLeaves[computedStateIndex]];
-
         const z = BigInt(42);
 
         const [c1r, c2r] = elGamalRerandomize(
-            newPublicKey.rawPubKey,
+            this.coordinatorKeypair.pubKey.rawPubKey,
             z,
             deactivatedKeyEvent.c1,
             deactivatedKeyEvent.c2,
         );
 
-        // TODO: check if asCircuitInputs should be used
         const nullifier = hash2([BigInt(deactivatedPrivateKey.asCircuitInputs()), salt]);
 
-        // TODO: set proper value for voiceCreditBalance
-        const voiceCreditBalance = BigInt(100)
+        // TODO: set proper value for newCreditBalance - take from input param?
+        const newCreditBalance = BigInt(5)
```

Comment:
Why the `newCreditBalance` is set to 5?

Reply:
Added a more elaborate TODO comment

Reply:
Ahh, I see, we need to take it as new input parameter, makes sense now. What are the barriers to implementing it right now?